README.md
Generic Gallery
===============

TYPO3 CMS Extension Generic Gallery (Extbase / Fluid rewrite)

_beta quality_



Roadmap
-------

* Test examples again after moving repo (re-check this whole thing)
* Release v1.0.0 by pushing a tag (Gizzle Github Hook)


ToDo
----
* Inline RTE fails sometimes, incorrect buttonset
* Ensure TYPO3 CMS 7.0 compatibility
* Fix inline @ToDo's


Future ideas
------------

* Add collection preview in BE
* Add TS type preview in BE
* Add unit tests
* Disable textItem position and width fields via TER or TS


Contribution
------------

Any help is appreciated. Please feel free to drop me a line, open issues or send pull requests.

It is possible to sponsor features and bugfixes!


Donation
--------

Please consider a donation: http://www.felixnagel.com/donate/

ChangeLog
Since version 1.0.0 please see:
https://github.com/fnagel/generic-gallery/commits

2014-09-13 Felix Nagel  <info@felixnagel.com>

	released v0.4.0 (TYPO3 migration required!)
	* removed: DAM support, now uses FAL
	* updated: now depends on TYPO3 6.0-6.2
	* improved: small refactoring, CGL and code clean up
	* updated: documentation and examples

2013-03-12 Felix Nagel  <info@felixnagel.com>

	* fixed: changelog (wrong version number in last release)

2013-03-09 Felix Nagel  <info@felixnagel.com>

	released v0.3.9 (no changes required)
	* fixed: php warning in in list module

2012-09-06 Felix Nagel  <info@felixnagel.com>

	released v0.3.8 (no changes required)
	* same version as 0.3.7 due to problems with TER upload

2012-09-06 Felix Nagel  <info@felixnagel.com>

	released v0.3.7 (no changes required)
	* removed: deprecated functions (TYPO3 6.0.0)
	* changed: TYPO3 4.7 is now max requirement (because of TYPO3 6.0 and FAL)

2012-08-25 Felix Nagel  <info@felixnagel.com>

	* fixed: exif data wont be displayed when switching to new, recommended EXT:svmetaextract
	* added: suggest svmetaextract
	* changed: files and inline code is now added using pageRenderer functions
	* changed: TYPO3 4.3 is now minimal requirement

2012-08-12 Felix Nagel  <info@felixnagel.com>

	* added: shuffle (random ordering) for images (thanks to Paints Multimedia for sponsoring, http://www.paints.de)

2012-06-04 Felix Nagel  <info@felixnagel.com>

	* fixed: BE search problems due to changes in TYPO3 4.6, for livesearch use gg_pictures and gg_content shortcuts

2012-05-24 Felix Nagel  <info@felixnagel.com>

	* improved: small changes to eID

2012-04-27 Felix Nagel  <info@felixnagel.com>

	released v0.3.6 (no changes required)
	* TYPO3 4.7.0 tested
	* added: TYPO3 4.1 dependency (needed for IRRE)
	* added: plugin.tx_genericgallery_pi1.baseWrap.wrap TS, see #33029, THX to Daniel K.

2011-12-30 Felix Nagel  <info@felixnagel.com>

	released v0.3.5 (no changes required)
	* improved: BE preview for single mode
	* fixed: small typo in BE
	* fixed: problem with RTE text and JS string escaping
	* improved: added sql_free_result()
	* added: custom marker documentation
	* fixed: deprecated functions

2011-11-07 Felix Nagel  <info@felixnagel.com>

	released v0.3.4 (changes might be required, see doc)
	* fixed: BE user group permisson problem
	* improved: BE preview
	* fixed: removed linebreaks (\n) in RTE text so its usable in JS strings

2011-09-07 Felix Nagel  <info@felixnagel.com>

	released v0.3.3 (changes might be required, see doc)
	* fixed: linking problem when using RTE description
	* fixed: typo in doc, thx to Renate R.
	* added: documentation about an issue with backend layout and templavoila_framework

2011-07-15 Felix Nagel  <info@felixnagel.com>

	* fixed: issue with endtime in single mode, see http://forumdev.typo3.org/t/214598/ and http://forge.typo3.org/issues/28298

2011-07-12 Felix Nagel  <info@felixnagel.com>

	released v0.3.2 (no changes required)
	* fixed: issue with single mode, thx to Herman Kremer
	* improved: IRRE record (single mode) title

2011-06-16 Felix Nagel  <info@felixnagel.com>

	released v0.3.1 (no changes required)
	* fixed: small issue with BE layout

2011-05-26 Felix Nagel  <info@felixnagel.com>

	released v0.3.0 (changes required)
	* added: you are now able to set backfalls for DAM fields
	* added: demo TS via static template
	* added: gallery mode displayed in BE preview
	* added: ajax demo
	* added: custom marker demo
	* changed: marker WIDTH and HEIGHT to IMAGE_WIDTH and IMAGE_HEIGHT
	* updated: documentation
	* improved: code quality
	* improved: extended demo with exif

2011-03-19 Felix Nagel  <info@felixnagel.com>

	released v0.2.0 (bugfix and feature release, no changes required)
	* fixed: replacing marker when no ROW and / or IMAGE_X  marker is defined
	* added: json and xml AJAX support
	* updated: documentation

2011-02-22 Felix Nagel  <info@felixnagel.com>

	released v0.1.2
	* fixed: improved BE user right management

2011-02-17 Felix Nagel  <info@felixnagel.com>

	released v0.1.1
	* fixed: small issue with BE layout

2011-02-16 Felix Nagel  <info@felixnagel.com>

	* first release to TER

2011-02-15 Felix Nagel  <info@felixnagel.com>

	* added localized uid marker
	* added busy noggin TV framework fallback
	* added option to enable / disable CMS layout in BE
	* updated demos to CDN ressources
	* updated: documentation

2011-01-28 Felix Nagel  <info@felixnagel.com>

	* fixed: problem with EXT:fl_realurl_image and imagesize()
	* improved: BE layout in mass mode

2011-01-23 Felix Nagel  <info@felixnagel.com>

	* added: documentation

2011-01-20 Felix Nagel  <info@felixnagel.com>

	* improved: TS markup

2011-01-19 Felix Nagel  <info@felixnagel.com>

	* added: title and link fields
	* added: BE preview layout
	* fixed: BE layout descriptions

2011-01-17 Felix Nagel  <info@felixnagel.com>

	* added: basic functionality

2011-01-14 Felix Nagel  <info@felixnagel.com>

	* initial start from user_ EXT (Paints Multimedia GmbH, http://paints.de)


tca.php
<?php
if (!defined('TYPO3_MODE')) die ('Access denied.');

$TCA['tx_generic_gallery_pictures'] = array(
	'ctrl' => $TCA['tx_generic_gallery_pictures']['ctrl'],
	'interface' => array(
		'showRecordFieldList' => 'hidden,starttime,endtime,tt_content_id,images'
	),
	'feInterface' => $TCA['tx_generic_gallery_pictures']['feInterface'],
	'columns' => array(
		'hidden' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config' => array(
				'type' => 'check',
				'default' => '0'
			)
		),
		'starttime' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xml:LGL.starttime',
			'config' => array(
				'type' => 'input',
				'size' => '8',
				'max' => '20',
				'eval' => 'date',
				'default' => '0',
				'checkbox' => '0'
			)
		),
		'endtime' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xml:LGL.endtime',
			'config' => array(
				'type' => 'input',
				'size' => '8',
				'max' => '20',
				'eval' => 'date',
				'checkbox' => '0',
				'default' => '0',
				'range' => array(
					'upper' => mktime(3, 14, 7, 1, 19, 2038),
					// 'lower' => mktime(0, 0, 0, date('m')-1, date('d'), date('Y'))
					'lower' => 0
				)
			)
		),
		'tt_content_id' => Array(
			'config' => Array(
				'type' => 'passthrough',
			)
		),
		'title' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_pictures.title',
			'config' => array(
				'type' => 'input',
				'size' => '30',
				'max' => '255',
				'checkbox' => '',
				'eval' => 'trim',
			)
		),
		'link' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_pictures.link',
			'config' => array(
				'type' => 'input',
				'size' => '15',
				'max' => '255',
				'checkbox' => '',
				'eval' => 'trim',
				'wizards' => array(
					'_PADDING' => 2,
					'link' => array(
						'type' => 'popup',
						'title' => 'Link',
						'icon' => 'link_popup.gif',
						'script' => 'browse_links.php?mode=wizard',
						'JSopenParams' => 'height=300,width=500,status=0,menubar=0,scrollbars=1'
					)
				)
			)
		),
		'images' => array(
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_pictures.images',
			'config' => \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::getFileFieldTCAConfig(
				'tx_generic_gallery_picture_single',
				array(
					'size' => 1,
					'maxitems' => 1,
					'minitems' => 1,
				),
				'jpg,gif,jpeg,png'
			)
		),
		'contents' => Array(
			'exclude' => 1,
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_pictures.contents',
			'config' => Array(
				'type' => 'inline',
				'foreign_table' => 'tx_generic_gallery_content',
				'foreign_field' => 'pictures_id',
				'appearance' => Array(
					'useSortable' => 1,
					'collapseAll' => 1,
					'expandSingle' => 1,
				),
				'maxitems' => 99,
			)
		),
	),
	'types' => array(
		'0' => array('showitem' => 'hidden;;1;;1-1-1, title;;2;;1-1-1, link, images')
	),
	'palettes' => array(
		'1' => array('showitem' => 'starttime, endtime'),
		'2' => array('showitem' => 'contents')
	)
);

$TCA['tx_generic_gallery_content'] = array(
	'ctrl' => $TCA['tx_generic_gallery_content']['ctrl'],
	'interface' => array(
		'showRecordFieldList' => 'hidden,bodytext,position'
	),
	'feInterface' => $TCA['tx_generic_gallery_content']['feInterface'],
	'columns' => array(
		'hidden' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config' => array(
				'type' => 'check',
				'default' => '0'
			)
		),
		'bodytext' => Array(
			'exclude' => 1,
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_content.bodytext',
			'config' => Array(
				'type' => 'text',
				'cols' => '48',
				'rows' => '5',
			),
			'defaultExtras' => 'richtext[*]:rte_transform[flag=rte_enabled|mode=ts]',
		),
		'position' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_content.position',
			'config' => Array(
				'type' => 'input',
				'size' => '10',
				'default' => '0,0',
			)
		),
		'width' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:generic_gallery/locallang_db.xml:generic_gallery_content.width',
			'config' => Array(
				'type' => 'input',
				'size' => '10',
				'default' => '0',
			)
		)
	),
	'types' => array(
		'0' => array('showitem' => 'hidden;;1;;1-1-1, bodytext')
	),
	'palettes' => array(
		'1' => array('showitem' => 'position, width')
	)
);

?>